packages/wds-event-tool/src/main/kotlin/org/webdevandsausages/events/controller/DeleteRegistration.kt

```diff
@@ -0,0 +1,83 @@
+package org.webdevandsausages.events.controller
+
+import arrow.core.Either
+import org.http4k.contract.ContractRoute
+import org.http4k.contract.bindContract
+import org.http4k.contract.div
+import org.http4k.contract.meta
+import org.http4k.core.HttpHandler
+import org.http4k.core.Method
+import org.http4k.core.Request
+import org.http4k.core.Response
+import org.http4k.core.Status
+import org.http4k.lens.Path
+import org.http4k.lens.string
+import org.webdevandsausages.events.Router
+import org.webdevandsausages.events.dto.ErrorCode
+import org.webdevandsausages.events.error.RegistrationCancellationError
+import org.webdevandsausages.events.handleErrorResponse
+import org.webdevandsausages.events.service.CancelRegistrationService
+
+object DeleteRegistration {
+    private val registrationToken = Path.string().of("token")
+
+    fun route(
+        deleteRegistration: CancelRegistrationService,
+        handleErrorResponse: handleErrorResponse
+    ): ContractRoute {
+
+        @Suppress("UNUSED_PARAMETER")
+        fun handleCancellation(token: String): HttpHandler = { req: Request ->
+
```

Comment:
This is pattern matching for errors is ok but kind of verbose. I have been thinking I will try to refactor at some point to use a http4k custom Filter. I think you should be able somehow to to pass the errors on and let a Filter wrap them. Not sure yet though how that works...